src/gpuarray_buffer_collectives.c

```diff
@@ -4,43 +4,58 @@
 #include "gpuarray/buffer_collectives.h"
 #include "gpuarray/error.h"
 
-int gpucomm_new(gpucomm* comm, gpucontext* ctx, const char* clique_id,
-                     int ndev, int rank, int* res) {
-  comm = NULL;
-  if (ctx->comm_ops == NULL)
+int gpucomm_new(gpucomm** comm, gpucontext* ctx,
+                gpucommCliqueId cliqueId, int ndev, int rank) {
+  if (ctx->comm_ops == NULL) {
+    *comm = NULL;
     return GA_UNSUPPORTED_ERROR;
-  return ctx->comm_ops->comm_new(comm, ctx, clique_id, ndev, rank, res);
+  }
+  return ctx->comm_ops->comm_new(comm, ctx, cliqueId, ndev, rank);
 }
 
-void gpucomm_free(gpucomm* comm) {
-  gpucomm_context(comm)->comm_ops->comm_free(comm);
+int gpucomm_free(gpucomm* comm) {
+  gpucontext* ctx = gpucomm_context(comm);
+  if (ctx->comm_ops == NULL)
+    return GA_COMM_ERROR;
```

Comment:
In case user calls this, while comm_ops are not supported (because he ignored the errors returned), should we let this cause a fault?